sdk-ext-lib/src/main/java/cash/z/ecc/sdk/model/WalletAddress.kt

```diff
@@ -0,0 +1,46 @@
+package cash.z.ecc.sdk.model
+
+sealed class WalletAddress(val address: String) {
+    class Unified private constructor(address: String) : WalletAddress(address) {
+        companion object {
+            suspend fun new(address: String): WalletAddress.Unified {
+                // https://github.com/zcash/zcash-android-wallet-sdk/issues/342
+                // TODO [#342]: refactor SDK to enable direct calls for address verification
+                return WalletAddress.Unified(address)
+            }
+        }
+    }
+
+    class Shielded private constructor(address: String) : WalletAddress(address) {
```

Comment:
Is this referring to Sapling Shielded addresses? 

Consider that Orchard is also shielded Zcash, so it could be confusing.